src/components/buttons/Send/SendFlow.tsx

```diff
@@ -0,0 +1,367 @@
+import ShareEcash from '@/components/views/ShareEcash';
+import SelectGift from '@/components/views/SelectGift';
+import { getAmountFromInvoice } from '@/utils/bolt11';
+import QRScanner from '@/components/views/QRScanner';
+import useWallet from '@/hooks/boardwalk/useWallet';
+import { useNumpad } from '@/hooks/util/useNumpad';
+import { useCashu } from '@/hooks/cashu/useCashu';
+import { useToast } from '@/hooks/util/useToast';
+import InputOptions from './InputOptions';
+import AmountInput from './AmountInput';
+import InputLud16 from './Lud16Input';
+import { useState } from 'react';
+
+type ActiveTab = 'ecash' | 'lightning';
+
+type MyState = {
+   activeTab: ActiveTab;
+   isProcessing: boolean;
+   isGiftMode: boolean;
+} & (
+   | {
+        step: 'input';
+        contact?: PublicContact;
+        paymentRequest?: PaymentRequest;
+     }
+   | {
+        step: 'lud16Input';
+        amount: number;
+        lud16: string;
+     }
+   | {
+        step: 'QRScanner';
+     }
+   | {
+        step: 'sendLightning';
+        invoice: string;
+        amount: number;
+        meltQuote?: MeltQuoteResponse;
+        lud16?: string;
+     }
+   | {
+        step: 'shareEcash';
+        token?: string;
+        txid?: string;
+        gift?: GiftAsset;
+        contact?: PublicContact;
+     }
+   | {
+        step: 'confirmPaymentRequest';
+        paymentRequest: PaymentRequest;
+        amount: number;
+     }
+   | {
+        step: 'confirmSendGift';
+        gift: GiftAsset;
+        contact?: PublicContact;
+     }
+   | {
+        step: 'selectContact';
+     }
+   | {
+        step: 'selectGift';
+        contact?: PublicContact;
+     }
+);
+
+const defaultState: MyState = {
+   step: 'input',
+   activeTab: 'ecash',
+   isProcessing: false,
+   isGiftMode: false,
+};
+
+interface SendFlowProps {
+   isMobile: boolean;
+   onReset: () => void;
+}
+
+const SendFlow = ({ isMobile, onReset }: SendFlowProps) => {
+   const [state, setState] = useState<MyState>(defaultState);
+
+   const { unitToSats, convertToUnit } = useExchangeRate();
+   const { sendGift, sendEcash } = useWallet();
+   const { isMintless } = useMintlessMode();
+   const { activeUnit } = useCashuContext();
+   const { getMeltQuote } = useCashu();
+   const { addToast } = useToast();
+   const showDecimal = activeUnit === Currency.USD;
+   const numpad = useNumpad({ showDecimal });
+   const { numpadValueIsEmpty, clearNumpadInput, numpadValue, numpadAmount } = numpad;
+
+   const resetState = () => {
+      setState(defaultState);
+      clearNumpadInput();
+      onReset();
+   };
+
+   const handleAmountInputSubmit = (input: number) => {
+      if (state.step !== 'input') return;
+
+      if (state.paymentRequest !== undefined) {
+         /* user entered an amountless payment request */
+         const newPR = state.paymentRequest;
+         setState(state => ({
+            ...state,
+            step: 'confirmPaymentRequest',
+            paymentRequest: newPR,
```

Comment:
The issue was this type error:
Property 'paymentRequest' does not exist on type 'SendFlowState'.
  Property 'paymentRequest' does not exist on type '{ activeTab: ActiveTab; isProcessing: boolean; isGiftMode: boolean; } & { step: "lud16Input"; amount: number; lud16: string; }'

I removed the callback to set the state directly instead of using previous state, hows that?

Reply:
ah I see. you can fix that by naming your inner variable differently so:
```
setState(currentState => ({
            ...currentState,
            step: 'confirmPaymentRequest',
            paymentRequest: state.paymentRequest,
            amount: input,
         }));
```

if you don't use callback it will capture the current value and that can cause issues if some other state change is doing changes at the same time

Reply:
still gives me this:
```
 types of property 'paymentRequest' are incompatible.
              Type 'PaymentRequest | undefined' is not assignable to type 'PaymentRequest'.
                Type 'undefined' is not assignable to type 'PaymentRequest'.
```

When I do:
```ts
 if (state.paymentRequest !== undefined) {
         /* user entered an amountless payment request */
         setState(currentState => ({
            ...currentState,
            step: 'confirmPaymentRequest',
            paymentRequest: state.paymentRequest,
            amount: input,
         }));
      }
 ```
 
 What if I just cast the type? Solves the type error:
 ```ts
  setState(currentState => ({
            ...currentState,
            step: 'confirmPaymentRequest',
            paymentRequest: state.paymentRequest as PaymentRequest,
            amount: input,
         }));
 ```

Reply:
something is wrong there. it shouldn't be showing that since there is `if (state.paymentRequest !== undefined)` check